Update index.md (#23861)
dotnet/docs: docs/csharp/programming-guide/arrays/index.md

- A jagged array is an array of arrays, and therefore its elements are reference types and are initialized to `null`.
- Arrays are zero indexed: an array with `n` elements is indexed from `0` to `n-1`.
- Array elements can be of any type, including an array type.

> - Array types are [reference types](../../language-reference/keywords/reference-types.md) derived from the abstract base type <xref:System.Array>. Since this type implements <xref:System.Collections.Generic.IList%601>, <xref:System.Collections.IEnumerable>, and <xref:System.Collections.Generic.IEnumerable%601>, you can use [foreach](../../language-reference/keywords/foreach-in.md) iteration on all arrays in C#.

Reviewer: This should call out single dimension arrays that implement the generic interfaces. (Not all arrays do).

```suggestion
- Array types are [reference types](../../language-reference/keywords/reference-types.md) derived from the abstract base type <xref:System.Array>. All arrays implement <xref:System.Collections.IList>, and <xref:System.Collections.IEnumerable>.  You can use [foreach](../../language-reference/keywords/foreach-in.md) iteration arrays in C# .  Since single-dimension arrays also implement <xref:System.Collections.Generic.IList%601>,  and <xref:System.Collections.Generic.IEnumerable%601>, 
```

> - Array types are [reference types](../../language-reference/keywords/reference-types.md) derived from the abstract base type <xref:System.Array>. All arrays implement <xref:System.Collections.IList>, and <xref:System.Collections.IEnumerable>.  You can use [foreach](../../language-reference/keywords/foreach-in.md) iteration arrays in C# .  Since single-dimension arrays also implement <xref:System.Collections.Generic.IList%601>,  and <xref:System.Collections.Generic.IEnumerable%601>, 

Reviewer: ```suggestion
- Array types are [reference types](../../language-reference/keywords/reference-types.md) derived from the abstract base type <xref:System.Array>. All arrays implement <xref:System.Collections.IList>, and <xref:System.Collections.IEnumerable>.  You can use [foreach](../../language-reference/keywords/foreach-in.md) iteration arrays in C# .  Since single-dimension arrays also implement <xref:System.Collections.Generic.IList%601>,  and <xref:System.Collections.Generic.IEnumerable%601>.
```

Author: Thanks for the suggestions and feedback. The added clarification and correction will be helpful to those using the documentation.
